src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Responsible for loading and saving tasks to memory.
+ */
+public class Storage {
+    private String dirName = "data/";
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = dirName + filePath;
+    }
+
+    /**
+     * Loads the tasks from the storage directory.
+     *
+     * @return {@code ArrayList<Task>} of the tasks stored in the storage directory.
+     * @throws DukeException If the tasks were not saved in appropriate format.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> result = new ArrayList<>();
+        try {
+            // Create the directory if it does not exist
+            File dir = new File(dirName);
+            if (!dir.exists()) {
+                dir.mkdirs();
+            }
+
+            File f = new File(filePath);
+            Scanner sc = new Scanner(f);
```

Comment:
I feel that f and sc could be renamed to more meaningful names as well rather than their abbreviations.